test/decompression.jl

```diff
@@ -0,0 +1,21 @@
+module TestCompression
+using Test
+using DataFrames
+
+# These must be loaded before RData
+using CodecBzip2
```

Comment:
```suggestion
using CodecBzip2, CodecXz
```
Do we really need it, though?
If we do, could you pls add a test that without this `using` the `load()` would throw `codec_misssing_msg`?

Reply:
We do need to load them in client code. But actually this isn't quite right, we need to have loaded the codecs _before_ calling `RData.load`, it doesn't actually matter if they are loaded after RData.

I've added a test that demonstrates this. But it still needs work as `FileIO` is spewing an [error message to the stderr](https://github.com/JuliaIO/FileIO.jl/blob/471dde68b1f7a3f465b829ae0c722b95ec26e7e1/src/error_handling.jl#L63) which I think we should silence because it makes it look like our tests are failing silently.

```
Error encountered while loading "/Users/danm/Work/NHANES.jl/dev/RData/test/data_v3/compressed_bzip2.rda".
Fatal error:
Error encountered while loading "/Users/danm/Work/NHANES.jl/dev/RData/test/data_v3/compressed_xz.rda".
Fatal error:
Test Summary:                                       | Pass  Total
When the optional compression codecs are not loaded |    2      2
```